mods/d2k/maps/atreides-03a/atreides03a.lua

```diff
@@ -0,0 +1,175 @@
+
+AtreidesBaseBuildings = { "barracks", "light_factory" }
+AtreidesUpgrades = { "upgrade.barracks", "upgrade.light" }
+
+wave = 0
+SendOrdos = function()
+	Trigger.AfterDelay(OrdosAttackDelay[Map.Difficulty], function()
+		if player.IsObjectiveCompleted(KillOrdos) then
+			return
+		end
+
+		wave = wave + 1
+		if wave > OrdosAttackWaves[Map.Difficulty] then
+			return
+		end
+
+		local units = Reinforcements.ReinforceWithTransport(ordos, "carryall.reinforce", OrdosReinforcements[Map.Difficulty][wave], OrdosPaths[1], { OrdosPaths[1][1] })[2]
+		Utils.Do(units, IdleHunt)
+
+		SendOrdos()
+	end)
+end
+
+MessageCheck = function(index)
+	return #player.GetActorsByType(AtreidesBaseBuildings[index]) > 0 and not player.HasPrerequisites({ AtreidesUpgrades[index] })
+end
+
+Tick = function()
+	if player.HasNoRequiredUnits() then
+		ordos.MarkCompletedObjective(KillAtreides)
+	end
+
+	if ordos.HasNoRequiredUnits() and not player.IsObjectiveCompleted(KillOrdos) then
+		Media.DisplayMessage("The Ordos have been anihilated!", "Mentat")
+		player.MarkCompletedObjective(KillOrdos)
+	end
+
+	if DateTime.GameTime % DateTime.Seconds(30) and HarvesterKilled then
+		local units = ordos.GetActorsByType("harvester")
+
+		if #units > 0 then
+			HarvesterKilled = false
+			ProtectHarvester(units[1])
+		end
+	end
+
+	if player.Resources > ToHarvest[Map.Difficulty] - 1 then
+		player.MarkCompletedObjective(GatherSpice)
+	end
+
+	if DateTime.GameTime % DateTime.Seconds(32) == 0 and (MessageCheck(1) or MessageCheck(2)) then
+		Media.DisplayMessage("Upgrade barracks and light factory to produce more advanced units.", "Mentat")
+	end
+
+	if HasPower and player.Resources > player.ResourceCapacity * 0.8 and DateTime.GameTime % DateTime.Seconds(32) == 0 then
+		Media.DisplayMessage(Messages[4], "Mentat")
+	end
+
+	UserInterface.SetMissionText("Harvested resources: " .. player.Resources .. "/" .. ToHarvest[Map.Difficulty], player.Color)
+end
+
+WorldLoaded = function()
+	ordos = Player.GetPlayer("Ordos")
+	player = Player.GetPlayer("Atreides")
+
+	InitObjectives()
+
+	Camera.Position = AConyard.CenterPosition
+
+	Trigger.OnAllKilled(OrdosBase, function()
+		Utils.Do(ordos.GetGroundAttackers(), IdleHunt)
+	end)
+
+	SendOrdos()
+	ActivateAI()
+
+	Trigger.AfterDelay(DateTime.Minutes(2) + DateTime.Seconds(30), function()
+		Reinforcements.ReinforceWithTransport(player, "carryall.reinforce", AtreidesReinforcements, AtreidesPath, { AtreidesPath[1] })
+	end)
+end
+
+InitObjectives = function()
+	Trigger.OnObjectiveAdded(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+
+	KillAtreides = ordos.AddPrimaryObjective("Kill all Atreides units.")
+	GatherSpice = player.AddPrimaryObjective("Harvest " .. tostring(ToHarvest[Map.Difficulty]) .. " Solaris worth of Spice.")
+	KillOrdos = player.AddSecondaryObjective("Eliminate all Ordos units and reinforcements\nin the area.")
```

Comment:
I added the text for the secondary objective(s) myself. Only the primary objective(s)are from the original.

Feel free to suggest another description. :)